File: hecho con python/app/models/profesor.py
from sqlalchemy import Column, Integer, String
from sqlalchemy.orm import relationship
from app.models.base import Base

class Profesor(Base):
    __tablename__ = "profesores"
    
    id = Column(Integer, primary_key=True, index=True, autoincrement=True)
    nombre = Column(String(100), nullable=False)
    tipo_identificacion = Column(String(10), nullable=False) # CC, TI, CE, Pasaporte
    numero_identificacion = Column(String(30), unique=True, index=True, nullable=False)
    especialidad = Column(String(100), nullable=False)
    
    # Relación uno a muchos con Curso
    cursos = relationship("Curso", back_populates="profesor", cascade="all, delete-orphan")

File: hecho con java/src/main/java/com/umanizales/notasya/models/Profesor.java
package com.umanizales.notasya.models;

import jakarta.persistence.*;
import lombok.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "profesores")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Profesor {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String nombre;

    @Column(name = "tipo_identificacion", nullable = false, length = 10)
    private String tipoIdentificacion;

    @Column(name = "numero_identificacion", nullable = false, unique = true, length = 30)
    private String numeroIdentificacion;

    @Column(nullable = false, length = 100)
    private String especialidad;

    @OneToMany(mappedBy = "profesor", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<Curso> cursos = new ArrayList<>();
}
